docs/contributors/release.md

```diff
@@ -186,25 +186,55 @@ If you don't have access to [make.wordpress.org/core](https://make.wordpress.org
+
+1. Ensure the WordPress `trunk` branch is open for enhancements. 
+2. Check out the last published Gutenberg release branch `git checkout release/x.x`
+3. Create a Pull Request from this branch targetting `wp/trunk`.
+4. Merge the Pull Request.
+
+Now, the branch is ready to be used to publish the npm packages.
+
+1. Check out the `wp/trunk` branch.
+2. Run the [package release process] but when asked for the version numbers to choose for each package, (assuming the package versions are written using this format `major.minor.patch`) make sure to bump at least the `minor` version number. For example, if the CHANGELOG of the package to be released indicates that the next unreleased version is `5.6.1`, choose `5.7.0` as a version.
+3. Update the `CHANGELOG.md` files of the published packages with the new released versions and commit to the `wp/trunk` branch.
+4. Cherry-pick the "Publish" (created by Lerna) and the CHANGELOG update commits into the `master` branch of Gutenberg.
+
+Now, the npm packages should be ready and a patch can be created and commited into WordPress `trunk`.
 
-1. Check out the last published Gutenberg release branch `git checkout release/x.x`
-2. Create a Pull Request from this branch into `g-minor`.
-3. Merge the branch.
 
 ### Minor WordPress Releases
 
-For minor releases, the critical fixes targeted for this WordPress Minor release should be cherry-picked into the `g-minor` branch one by one in their chronological order.
+The following workflow is needed when bug fixes or security releases need to be backported into WordPress Core. This can happen in a few use-cases:
+
+ - During the `beta` and the `RC` period of the WordPress release cycle.
+ - For WordPress minor releases and WordPress security releases (example `5.1.1`).
+
+1. Cherry-pick
+2. Check out the last published Gutenberg release branch `git checkout release/x.x`
+3. Create a Pull Request from this branch targetting the WordPress related major branch (Example `wp/5.2`).
+4. Merge the Pull Request.
+
+Now, the branch is ready to be used to publish the npm packages.
+
+1. Check out the WordPress branch used before (Example `wp/5.2`).
+2. Run the [package release process] but when asked for the version numbers to choose for each package, (assuming the package versions are written using this format `major.minor.patch`) make sure to bump only the `patch` version number. For example, if the last published package version for this WordPress branch was `5.6.0`, choose `5.6.1` as a version.
+
+**Note:** For WordPress `5.0` and WordPress `5.1`, a different release process was used. This means that when choosing npm package versions targetting these two releases, you won't be able to use the next `patch` version number as it may have been already used. You should use the "metadata" modifier for these. For example, if the last published package version for this WordPress branch was `5.6.1`, choose `5.6.1+patch.1` as a version.
```

Comment:
This will be problematic for other projects using our _packages_, it is not semver friendly at all.

See also my above comment https://github.com/WordPress/gutenberg/pull/14136/files#r260688733

Reply:
What alternatives do we have?

Reply:
Continue to use semantic versioning:

If WordPress `5.1.1` is planned due to a security issue with the `@wordpress/editor` package which is at the time of writing is currently version `9.0.11` then the `@wordpress/editor` package is patched, updated, and published as version `9.1.0`.

That version `9.1.0` version of `@wordpress/editor` is then backported into both the `wp/5.0` and `wp/5.1` branches

Reply:
The thing is `9.1.0` might already exist because the `wp.5.2` might have already included it, so how do you go about patching the old versions without also adding features from the new versions?

Reply:
For the past you cannot, the deprecations in the _packages_ will have to support the previous versions of WordPress per the Gutenberg deprecation policy

Reply:
I'm not talking about breaking any compatibility here. I'm taking about:

 - `WP 5.0.3` includes the editor package with `9.0.11`
 - `WP 5.1` includes the editor package with version `9.0.12`
 - `Trunk` includes the editor package with version `9.1.0`

(These are random numbers to illustrate)

A security issue is discovered and concerns all these packages, WordPress has to ship a security release for both `5.0`, `5.1` and also fix the issue in trunk. How do we proceed?

Contraints:

 - We want to include only this security fix for the previous versions, we don't want to add features that have been built for future versions.

Which versions do you publish and include in the WordPress releases `5.0.4`, `5.1.1` and trunk?

Reply:
Based on the above https://github.com/WordPress/gutenberg/pull/14136#discussion_r260710630 knowing that we can backport the fixes going for the option of bumping the packages _minor_ version number before release would allow us to avoid the above scenario...

As the WP 5.0 branch would have v9.0.x of the package, before releasing WP 5.1 the package _minor_ is bumped to v9.1.x, this allows for multiple 9.0.x release to be backport puiblished for WP 5.0.x's, eg, `9.0.11` for WP `5.0.3`, `9.0.12` for WP `5.0.4`, `9.0.13` for WP `5.0.5 ` etc etc, and for WP 5.1.x, `9.1.1`, `9.1.2`, `9.1.3` and on are available, bump the _minor_ again for WP 5.2 and `9.2.x` range is available.

For projects other than WordPress using our packages, a note on all the repos that a _minor_ bump to all packages occurs when WP publishes a new version should document for the most part as to why we are slightly breaking semver on our packages, occasionly.

Reply:
> `9.0.12` for WP `5.0.4`

In my example, this is not possible because `9.0.12` already existed and is used in `5.1` and we don't have room to add fixes between `9.0.11` and `9.0.12`, that's why this is a special case using "metadata". I know that it's not perfect and that it's not "full semver" as npm will consider these versions as the same unless you explicitely target one but I don't think we have choices here because this scenario already happened, in this part, I'm trying to find a way to solve it when we face it.

Reply:
Yes, I realise that, what I'm saying is we should do as you suggested above, and in the above scenario before WP 5.1 was shipped we would have bumped the editor package from the current `9.0.11` to `9.1.0`, thus WP 5.1 would have shipped with `9.1.0` and we'd have the remaining `9.0.x` versions for as many WP 5.0.x releases required over the coming years, WP 5.0.55 would use `9.0.66` 😏